### ISSUE TITLE
invalid compiler args for glbinding (cmake) dependency are set

### ISSUE THREAD

**Issue by spholz** [mesonbuild/meson#10288]
**Describe the bug**
Meson generates invalid compiler args (a `'>'`) when configuring a project using glbinding as a cmake dependency. (it also adds `-std=c++11` as a compiler arg changing the c++ standard in use)
This causes g++ to fail:
```
ninja: Entering directory `/tmp/test/builddir'
[1/2] Compiling C++ object example.p/test.cpp.o
FAILED: example.p/test.cpp.o 
sccache c++ -Iexample.p -I. -I.. -I/usr/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++14 -O0 -g -DSYSTEM_LINUX -pthread -std=c++11 '>' -MD -MQ example.p/test.cpp.o -MF example.p/test.cpp.o.d -o example.p/test.cpp.o -c ../test.cpp
c++: warning: >: linker input file unused because linking not done
c++: error: >: linker input file not found: No such file or directory
ninja: build stopped: subcommand failed.
```

After doing a quick `git bisect` this issue seems to be introduced in 99aae9b4dfc7f1ab6469adee718d54d9afd45097.
Somehow this line in the glbinding cmake file (`/usr/share/glbinding/cmake/glbinding/glbinding-export.cmake`) is parsed wrong?
```cmake
  INTERFACE_COMPILE_OPTIONS "\$<\$<PLATFORM_ID:Darwin>:;-pthread;>;\$<\$<VERSION_LESS:3.14.4,3.1>:;-std=c++11;>"
```

**To Reproduce**
meson.build:
```meson
project('example', 'cpp',
  version : '0.1',
  default_options : ['warning_level=3', 'cpp_std=c++14'])

executable('example',
           'test.cpp',
           dependencies : [
              dependency('glbinding')],
           install : true)
```
test.cpp:
```c++
int main() { }
```

**Expected behavior**
The glbinding cmake file is parsed correctly

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)?
native build
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.)
Arch Linux
* what Python version are you using e.g. 3.8.0
Python 3.10.4
* what `meson --version`
0.62.0
* what `ninja --version` if it's a Ninja build
1.10.2

**Comment by eli-schwartz** [comment]
> Meson generates invalid compiler args (a `'>'`) when configuring a project using glbinding as a cmake dependency.

cmake generator expressions are, unfortunately, madness. :(

> it also adds `-std=c++11` as a compiler arg changing the c++ standard in use

Unfortunately this seems to be the fault of the cmake definition...

**Comment by spholz** [comment]
`-std=c++11` isn't set as a compile arg before commit 99aae9b4dfc7f1ab6469adee718d54d9afd45097.
is setting `-std=c++11` the correct behavior for this cmake definition?

**Comment by mensinda** [comment]
Should be fixed with #10289. The problem was that the generator expression parser now runs after the input strings are split into lists by `;`.

**Comment by spholz** [comment]
https://github.com/mesonbuild/meson/pull/10289 seems to have fixed the bug. Thanks for fixing this so quickly!
